### PR TITLE
Indicator mass index

### PR DESCRIPTION
**Pull request by Jay-Jay-D**
Mass indicator index implementation #1069

### CODE REVIEW

File: Indicators/MassIndex.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>
+        /// A new value for this indicator
+        /// </returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            _highLow.Update(new IndicatorDataPoint
```
Comment (mchandschuh) [inline Indicators/MassIndex.cs]: Does this need to be an indicator?

Comment (Jay-Jay-D) [reply Indicators/MassIndex.cs]: No, is a remnant of a previous version. I'll fix it now.

The first attempt I made was [this](https://github.com/Jay-Jay-D/LeanSTP/blob/df74ac295c2f0400bdb8d6293307970675ba7874/Tests/Indicators/MassIndexIndicatorTests.cs), but fails the `ComparesAgainstExternalDataAfterReset` test.

File: Indicators/MassIndex.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    /// <summary>
+    /// The Mass Index uses the high-low range to identify trend reversals based on range expansions.
+    /// In this sense, the Mass Index is a volatility indicator that does not have a directional
+    /// bias. Instead, the Mass Index identifies range bulges that can foreshadow a reversal of the
+    /// current trend. Developed by Donald Dorsey.
+    /// </summary>
+    /// <seealso cref="QuantConnect.Indicators.IndicatorBase{QuantConnect.Data.Market.TradeBar}"/>
+    public class MassIndex : IndicatorBase<TradeBar>
+    {
+        private readonly int _sumPeriod;
```
Comment (Martin-Molinero) [inline Indicators/MassIndex.cs]: doesn't Sum make the period available?

Comment (Jay-Jay-D) [reply Indicators/MassIndex.cs]: Actually yes, thanks!

**Pull request merged**